Realm/RLMQueryUtil.mm

```diff
@@ -823,8 +845,57 @@ void validate_and_extract_between_range(id value, RLMProperty *prop, id *from, i
 
 #pragma mark Link Constraints
 
+void QueryBuilder::add_memberwise_equality_constraint(const ColumnReference& column, RLMObjectBase *obj) {
+    for (RLMProperty *property in obj->_objectSchema.properties) {
```

Comment:
When we're comparing an object link to a thing mapped to an object, the RHS will always be an unmanaged object and match nothing. This isn't very useful, so instead of comparing the link it's comparing each property in the object to the corresponding property in the RHS. This is possibly the thing we want to be doing for all embedded objects, but that'd be a v11 change.